### PR TITLE
fix: remove options.model overwrite on bulkUpdate

### LINKED ISSUE #11364: Update maps returned instance to the wrong model
## Issue Description

When multiple models are wired up to the same table. The model's static update method can incorrectly map the returned instance to the wrong model

### What are you doing?
```
class Foo extends Model {}
Foo.init({ /* bla */ }, {
  tableName: 'table_with_discriminator',
  sequelize
})

class Bar extends Model {}
Bar.init({ /* bla */ }, {
  tableName: 'table_with_discriminator',
  sequelize
})


const updates = await Bar.update(
      { prop: 'value' },
      {
        where: { id: 'xxx },
        returning: true
      }
    );
const updatedInstance = updates[1][0]; // this is an instance of Foo
```

### What do you expect to happen?

The updated instance should be an instance of Bar

### What is actually happening?

The updated instance is an instance of Foo

### Additional context
The root cause if this line:
https://github.com/sequelize/sequelize/blob/master/lib/query-interface.js#L1040
Which is called from the update method here:
https://github.com/sequelize/sequelize/blob/master/lib/model.js#L3219

The returned instances are being mapped to a model based on the underlying table name.  The instance should be mapped to the same model used to call the update method

### Environment

- Sequelize version: 5.16.0
- Node.js version: 10.15.3
- Operating System: Mac

### How does this problem relate to dialects?

- [X] I think this problem happens regardless of the dialect.
- [ ] I think this problem happens only for the following dialect(s): 
- [ ] I don't know, I was using PUT-YOUR-DIALECT-HERE, with connector library version XXX and database version XXX

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [X] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

### PR DESCRIPTION
**Pull request by BigsonLvrocha**
## Pull Request Checklist

- [x] Have you added new tests to prevent regressions?
- [x] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

Closes #11364

This is a way that keeps the current behaviour of bulkUpdate while fixing the unexpected behaviour when calling `.update` when there are 2 or more models defined for the same table 

## Todos

- [x] #1 add test

### CODE REVIEW

**Review by ephys — APPROVED** [review]
This looks good to me

File: test/unit/model/update.test.js [hunk 1]
```diff
@@ -53,4 +53,56 @@ describe(Support.getTestDialectTeaser('Model'), () => {
...
+      }, {
+        tableName: 'model_table',
+      });
+
+      this.Model2 = current.define('Model2', {
+        value: DataTypes.INTEGER,
+        name: DataTypes.STRING,
+      }, {
+        tableName: 'model_table',
+        globalScope: {
```
Comment (ephys) [inline test/unit/model/update.test.js:73]: just one note though: we don't have a globalScope option. Did you mean defaultScope? Either way it's not stricly necessary for this test :)

**Review by ephys — CHANGES_REQUESTED** [review]

File: test/unit/model/update.test.js (hunk 1 above)
Comment (BigsonLvrocha) [reply test/unit/model/update.test.js]: oops, my mistake, it's there to make an example of an use case.
fixed in https://github.com/sequelize/sequelize/pull/15231/commits/af63e3d6a5dbb47d31d1ebabd994fec75aea211d

File: test/unit/model/update.test.js [hunk 2]
```diff
@@ -53,4 +53,56 @@ describe(Support.getTestDialectTeaser('Model'), () => {
...
+      this.Model2 = current.define('Model2', {
+        value: DataTypes.INTEGER,
+        name: DataTypes.STRING,
+      }, {
+        tableName: 'model_table',
+        defaultScope: {
+          where: {
+            isModel2: true,
+          },
+        },
```
Comment (WikiRik) [inline test/unit/model/update.test.js:77]:
```suggestion
```

**Comment by BigsonLvrocha** [comment]
sorry for the span on request, I did not know only one review was requested per pr

**Review by WikiRik — APPROVED** [review]

**Comment by WikiRik** [comment]
Thanks for this PR!

**Comment by BigsonLvrocha** [comment]
Will this fix go into v6?
Can it go in there too?

**Comment by BigsonLvrocha** [comment]
@WikiRik 
Will this change go into v6 too?
What should I do for the fix to get there?

**Comment by WikiRik** [comment]
Good question! Just open a PR just like this one, but into the `v6` branch. We'll review it and then it will land in the next v6 release

**Pull request merged**